rust/agama-server/src/agama-web-server.rs

```diff
@@ -100,30 +98,53 @@ struct ServeArgs {
 }
 
 impl ServeArgs {
-    /// Builds an SSL acceptor using a provided SSL certificate or generates a self-signed one
-    fn ssl_acceptor(&self) -> Result<SslAcceptor, openssl::error::ErrorStack> {
-        let mut tls_builder = SslAcceptor::mozilla_modern_v5(SslMethod::tls_server())?;
+    /// Returns true of given path to certificate points to an existing file
+    fn valid_cert_path(&self) -> bool {
+        self.cert.clone().is_some_and(|c| Path::new(&c).exists())
```

Comment:
```suggestion
        self.cert.as_ref().is_some_and(|c| Path::new(&c).exists())
```

There is not need to clone the value in this case.

Reply:
you know ... if you learned to use hammer .... ;-)